### PR TITLE
Rewrite BaselineAnalyzer::analyze

### PR DESCRIPTION
**Pull request by szepeviktor**
@staabm Would you welcome a PR that ends up like this?

### CODE REVIEW

**Comment by staabm** [comment]
Thanks for the suggestion.

 I think the code as is on master, would benefit from refactoring and could be improved for readability.

I don‘t like the suggested change though… it feels like shortening but without impeoving readability

File: lib/BaselineAnalyzer.php [hunk 1]
```diff
@@ -24,30 +24,14 @@ public function analyze(): AnalyzerResult
          * @var BaselineError $baselineError
          */
         foreach ($this->baseline->getIgnoreErrors() as $baselineError) {
-            $errorMessage = $baselineError->message;
-
-            if (str_contains($errorMessage, ' deprecated class ') || str_contains($errorMessage, ' deprecated method ')) {
```
Comment (staabm) [inline lib/BaselineAnalyzer.php:29]: IMO we could extract this if conditions into a `countDeprecation($errorMessage)`, but let the method here still work on the result of said method and adjust `$result->deprecations` here at the call-site.

File: lib/BaselineAnalyzer.php [hunk 2]
```diff
@@ -24,30 +24,14 @@ public function analyze(): AnalyzerResult
...
-                }
-            } elseif (str_contains($errorMessage, 'PHPDoc tag ')) {
-                $result->invalidPhpdocs += $baselineError->count;
-            } elseif (str_contains($errorMessage, ' not found')) {
-                preg_match('/Instantiated class .+ not found/', $errorMessage, $matches);
-                if ($matches) {
-                    $result->unknownTypes += $baselineError->count;
-                }
-
-            } elseif (str_contains($errorMessage, 'on an unknown class') || str_contains($errorMessage, 'has invalid type unknown') || str_contains($errorMessage, 'unknown_type as its type')) {
```
Comment (staabm) [inline lib/BaselineAnalyzer.php:44]: `countUnknownTypes()`

File: lib/BaselineAnalyzer.php [hunk 3]
```diff
@@ -24,30 +24,14 @@ public function analyze(): AnalyzerResult
...
-                $result->unknownTypes += $baselineError->count;
-            } elseif (str_contains($errorMessage, 'Anonymous variable')) {
```
Comment (staabm) [inline lib/BaselineAnalyzer.php:46]: `countAnonymousVariables()`

**Comment by szepeviktor** [comment]
All right.

File: lib/BaselineAnalyzer.php (hunk 1 above)
Comment (szepeviktor) [reply lib/BaselineAnalyzer.php:29]: Are you thinking about this?
```php
$result->deprecations += $this->countDeprecation();
$result->classesComplexity += $this->countClassesComplexity();
...
```
Removed `$errorMessage` as it comes from a class property.

Comment (staabm) [reply lib/BaselineAnalyzer.php:29]: Exactly

**Pull request closed without merging**